allennlp/training/trainer.py

```diff
@@ -552,7 +555,7 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
                         '{0}.{1}'.format(epoch, time_to_str(int(last_save_time))), [], is_best=False
                 )
 
-        return self._get_metrics(train_loss, batches_this_epoch, reset=True)
+        return self._get_metrics(train_loss, batches_this_epoch, reset=True), peak_cpu_usage, gpu_usage
```

Comment:
Instead of adding two return values here, just add them to the metrics dictionary directly:

```python
metrics = self._get_metrics(...)
metrics['cpu_memory'] = peak_cpu_usage
metrics['gpu_memory'] = gpu_usage
return metrics
```